### PR TITLE
Add docs and examples for sea ice production tasks

### PR DESCRIPTION
**Pull request by darincomeau**
Adding documentation and examples to https://github.com/MPAS-Dev/MPAS-Analysis/pull/907

### CODE REVIEW

**Comment by darincomeau** [comment]
@xylar I'm not sure how to get the docs to generate off a branch, so I haven't been able to verify/review the output here.

**Comment by xylar** [comment]
Thanks for the PR, @darincomeau!

>  I'm not sure how to get the docs to generate off a branch, so I haven't been able to verify/review the output here.

If you go into `docs` with your `mpas_dev` environment and run:
```
make html
```
then you should be able to build the docs on your laptop (or an HPC machine).  Then, open:
```
_build/html/index.html
```
in a browser to see the results.  If any red text comes up, we'd want to try to fix it.  Some of it may already be fixed in #919, though.

**Comment by xylar** [comment]
You could rebase if you want, since I just merged #919.  Then, hopefully you'll see a clean build.

**Comment by darincomeau** [comment]
Here's the documentation built from this branch on LCRC, so the LCRC header kinda gets in the way, but the new tasks show up:

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/add-seaice-production-docs/html/

But during the build, I get a huge number of red `WARNING: document isn't included in any toctree` that seems unrelated?

**Comment by xylar** [comment]
Perfect!  I don't know what the deal is with those "document isn't included" warnings but, no, don't worry about those.  This looks great!

File: docs/users_guide/tasks/climatologyMapSeaIceMeltingSH.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
+.. _task_climatologyMapSeaIceMeltingSH:
+
+climatologyMapSeaIceMeltingSH
+==========================
```
Comment (xylar) [inline docs/users_guide/tasks/climatologyMapSeaIceMeltingSH.rst:4]:
```suggestion
============================
```

File: docs/users_guide/tasks/climatologyMapSeaIceProductionNH.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
+.. _task_climatologyMapSeaIceProductionNH:
+
+climatologyMapSeaIceProductionNH
+==========================
```
Comment (xylar) [inline docs/users_guide/tasks/climatologyMapSeaIceProductionNH.rst:4]:
```suggestion
================================
```

File: docs/users_guide/tasks/climatologyMapSeaIceProductionSH.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
+.. _task_climatologyMapSeaIceProductionSH:
+
+climatologyMapSeaIceProductionSH
+==========================
```
Comment (xylar) [inline docs/users_guide/tasks/climatologyMapSeaIceProductionSH.rst:4]:
```suggestion
================================
```

File: docs/users_guide/tasks/climatologyMapSeaIceMeltingNH.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
+.. _task_climatologyMapSeaIceMeltingNH:
+
+climatologyMapSeaIceMeltingNH
+==========================
```
Comment (xylar) [inline docs/users_guide/tasks/climatologyMapSeaIceMeltingNH.rst:4]:
```suggestion
============================
```

**Review by xylar — CHANGES_REQUESTED** [review]
Just 4 really pedantic changes that sphinx tends to complain about.

**Review by xylar — APPROVED** [review]

**Comment by darincomeau** [comment]
ok great - thanks!

**Pull request merged**